NuGet/NuGet.Client
# Add support for verifying signed package against Repository Signature Info

## Bug
Fixes: https://github.com/NuGet/Home/issues/6700
Regression: No

## Fix
Details: This PR adds the following - 

1.  Updates SignedPackageVerifierSettings to contain 4 additional fields -
     - `bool AllowNoRepositoryCertificateList`
     - `bool AllowNoClientCertificateList`
     - `IReadOnlyList<VerificationAllowListEntry> RepositoryCertificateList`
     - `IReadOnlyList<VerificationAllowListEntry> ClientCertificateList`
     The new fields allow us to pass client and repository allow lists to `AllowListVerificationProvider`
2. Refactor `ISignatureVerificationProvider` to accept allow lists via `SignedPackageVerifierSettings`
3. Add `RepositorySignatureInfoUtility` to process `RepositorySignatureInfo` and generate effective `SignedPackageVerifierSettings`.
4. Update `PackageExtractor` to check if the download source has `RepositorySignatureInfo` and use that to verify signed packages.
5. Supporting refactoring and tests.


## Testing/Validation
Tests Added: Yes
